concent_api/core/validation.py

```diff
@@ -373,3 +369,36 @@ def validate_file_transfer_token(file_transfer_token: message.concents.FileTrans
     categories = [file_info['category'] for file_info in file_transfer_token.files]
     if len(set(categories)) != len(categories):
         raise FileTransferTokenError("'category' field must be unique across FileInfo list.", ErrorCode.MESSAGE_FILES_CATEGORY_NOT_UNIQUE)
+
+
+def validate_secure_hash_algorithm(checksum: str):
+    if not isinstance(checksum, str):
+        raise HashingAlgorithmError(
+            "'checksum' must be a string.",
+            ErrorCode.MESSAGE_FILES_CHECKSUM_WRONG_TYPE
+        )
+
+    if len(checksum) == 0 or checksum.isspace():
+        raise HashingAlgorithmError(
+            "'checksum' cannot be blank or contain only whitespace.",
+            ErrorCode.MESSAGE_FILES_CHECKSUM_EMPTY
+        )
+
+    if ":" not in checksum:
+        raise HashingAlgorithmError(
+            "'checksum' must consist of two parts separated with a semicolon.",
+            ErrorCode.MESSAGE_FILES_CHECKSUM_WRONG_FORMAT
+        )
+
+    if not checksum.split(":")[0] == set(HashingAlgorithm._value2member_map_).pop():  # type: ignore
```

Comment:
this is really bad way of checking if algorithm written in part of the checksum is supported. Why not just sth like:
`if checksum.split(":")[0] not in [e.value for e in HashingAlgorithm]`?
Or you can even add a static method to `HashingAlgorithm`:
```python
@unique
class HashingAlgorithm(Enum):
    SHA1 = 'sha1'

   def values():
       return [a.value for a in HashingAlgorithm]
```
and then check it like: ``if checksum.split(":")[0] not in HashingAlgorithm.values()`

Reply:
Done